components/scream/src/share/field/field_group.hpp

```diff
@@ -102,12 +104,43 @@ struct FieldGroup {
 
   FieldGroup (const FieldGroup&) = default;
 
+  // Allows to copy a non-const group into a const one.
+  template<typename RT,
+           typename = typename std::enable_if<
+               not std::is_same<RT,RealType>::value
+               and std::is_const<RealType>::value
+             >::type
+           >
+  FieldGroup (const FieldGroup<RT>& src) {
+    *this = src;
+  }
+
+  FieldGroup<typename const_field_type::RT> get_const () const {
+    return FieldGroup<typename const_field_type::RT>(*this);
+  }
+
+  template<typename RT,
+           typename = typename std::enable_if<
+               not std::is_same<RT,RealType>::value
+               and std::is_const<RealType>::value
+             >::type
+           >
+  FieldGroup<RealType>& operator= (const FieldGroup<RT>& src) {
+    m_info = src.m_info;
+    copy_fields<RT> (src);
+  }
+
   const std::string& grid_name () const {
-    EKAT_REQUIRE_MSG(m_fields.size()>0,
+    EKAT_REQUIRE_MSG(m_fields.size()>0 || m_bundle,
```

Comment:
Doesn't this predicate mean that `EKAT_REQUIRE_MSG` will report the error below if `m_bundle` is set? If so, what does it mean that we're checking `m_bundle` on line 137?

Reply:
Uhm, it reports the error if neither of the conditions pass. The grid name is pulled from any of the fields in the group, or from the bundled field. If neither are set, we can't retrieve the grid name. Since a group may have `m_fields` set, but not the bundle, or viceversa, at 137 we check which one is set, and grab the grid name from the first one that is available.

Reply:
Got it. Thanks!